tests/backup/backup_restore_basic_test.go

```diff
@@ -3712,3 +3713,327 @@ var _ = Describe("{DeleteS3ScheduleAndCreateNfsSchedule}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, s3CloudCredName, s3CloudCredUID, ctx)
 	})
 })
+
+// checkBackupObjectForNonExpectedNS checks if namespaces like kube-system and px namespace
+// is backed up or not
+func checkBackupObjectForNonExpectedNS(ctx context.Context, backupName string) {
+
+	var namespacesToSkip = []string{"kube-system", "kube-node-lease", "kube-public"}
+
+	// Fetch a list of backups
+	backupUID, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
+	dash.VerifyFatal(err, nil, "Fetch backup UID")
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   backupUID,
+		OrgId: orgID,
+	}
+	backupResponse, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+	dash.VerifyFatal(err, nil, "Fetch backup inspect object")
+	backupNamespaces := backupResponse.GetBackup().Namespaces
+
+	err = SetDestinationKubeConfig()
+	dash.VerifyFatal(err, nil, "Switch cluster context")
+	k8sCore := core.Instance()
+	allServices, err := k8sCore.ListServices("", metav1.ListOptions{})
+	dash.VerifyFatal(err, nil, "Get list of services")
+	for _, svc := range allServices.Items {
+		if svc.Name == "portworx-service" {
+			namespacesToSkip = append(namespacesToSkip, svc.Namespace)
+		}
+	}
+	for _, namespace := range backupNamespaces {
+		for _, namespacetoskip := range namespacesToSkip {
+			dash.VerifyFatal(namespacetoskip == namespace, false, fmt.Sprintf("Expected namespace %s shouldn't be present in backup", namespace))
+		}
+	}
+	err = SetSourceKubeConfig()
+	log.FailOnError(err, "Switching context to source cluster failed")
+}
+
+// getNamespaceAge gets the namespace age of all the namespaces on the cluster
+func getNamespaceAge() map[string]time.Time {
+	var namespaceAge = make(map[string]time.Time)
+	err := SetDestinationKubeConfig()
+	dash.VerifyFatal(err, nil, "Failed to switch cluster context")
+	k8sCore := core.Instance()
+	allNamespaces, err := k8sCore.ListNamespaces(make(map[string]string))
+	for _, namespace := range allNamespaces.Items {
+		namespaceAge[namespace.ObjectMeta.GetName()] = namespace.ObjectMeta.GetCreationTimestamp().Time
+	}
+	err = SetSourceKubeConfig()
+	log.FailOnError(err, "Switching context to source cluster failed")
+	return namespaceAge
+}
+
+// checkStatusOfNamespaces checks the status of namespaces on clusters where the restore was done
+func checkStatusOfNamespaces(oldNamespaceAge map[string]time.Time) {
```

Comment:
I feel we should take services, namespaces as input rather than switching clusters internally.

Reply:
How about we return error?

Reply:
The intend of these functions is to reduce code duplication in case of the specific case, returning error and again adding dash.verifyfatel will come back to the same problem of redundant code. Yes it would be 3-4 redundant lines then some 10-20 lines but I feel this way it's much more cleaner.